x/foundation/keeper/member.go

```diff
@@ -16,7 +16,10 @@ func validateMetadata(metadata string, config foundation.Config) error {
 
 func (k Keeper) UpdateDecisionPolicy(ctx sdk.Context, policy foundation.DecisionPolicy) error {
 	info := k.GetFoundationInfo(ctx)
-	info.SetDecisionPolicy(policy)
+	err := info.SetDecisionPolicy(policy)
+	if err != nil {
```

Comment:
You can use this one liner, because it returns only `error`.
```suggestion
	if err := info.SetDecisionPolicy(policy); err != nil {
```
It has the great advantage of not creating a new variable (outside of `if`). Then, you may revert the changes caused by the naming collision in variables.